File: api/tabung-api/src/main/java/my/emasjid/tabungapi/entity/Kutipan.java
package my.emasjid.tabungapi.entity;

import org.hibernate.annotations.Formula;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Data
@Entity
@Table(name = "kutipan")
public class Kutipan {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "tabung_id", referencedColumnName = "id")
    private Tabung tabung;

    private Integer total1c;
    private Integer total5c;
    private Integer total10c;
    private Integer total20c;
    private Integer total50c;
    private Integer total1d;
    private Integer total5d;
    private Integer total10d;
    private Integer total20d;
    private Integer total50d;
    private Integer total100d;

    @Formula(value = "(total1c * 0.01 + total5c * 0.05 + total10c * 0.1 + total20c * 0.2 + total50c * 0.5 + " +
            "total1d + total5d * 5 + total10d * 10 + total20d * 20 + total50d * 50 + total100d * 100)")
    private Long total;
    
    private Long createDate;
}

File: api/tabung-api-go/model/kutipan.go
package model

type Kutipan struct {
	Id         int64   `gorm:"column:id;primaryKey" json:"id"`
	TabungId   int64   `gorm:"column:tabung_id" json:"-"`
	Tabung     Tabung  `gorm:"foreignKey:TabungId" json:"tabung"`
	Total1c    int     `gorm:"column:total1c" json:"total1c"`
	Total5c    int     `gorm:"column:total5c" json:"total5c"`
	Total10c   int     `gorm:"column:total10c" json:"total10c"`
	Total20c   int     `gorm:"column:total20c" json:"total20c"`
	Total50c   int     `gorm:"column:total50c" json:"total50c"`
	Total1d    int     `gorm:"column:total1d" json:"total1d"`
	Total5d    int     `gorm:"column:total5d" json:"total5d"`
	Total10d   int     `gorm:"column:total10d" json:"total10d"`
	Total20d   int     `gorm:"column:total20d" json:"total20d"`
	Total50d   int     `gorm:"column:total50d" json:"total50d"`
	Total100d  int     `gorm:"column:total100d" json:"total100d"`
	Total      float64 `gorm:"-" json:"total"`
	CreateDate int64   `gorm:"column:create_date" json:"createDate"`
}

func (Kutipan) TableName() string {
	return "kutipan"
}
